stdlib/REPL/test/repl.jl

```diff
@@ -914,15 +914,26 @@ let ends_with_semicolon = REPL.ends_with_semicolon
     @test !ends_with_semicolon("")
     @test ends_with_semicolon(";")
     @test !ends_with_semicolon("a")
+    @test !ends_with_semicolon("a # asdf ;")
+    @test ends_with_semicolon("""a * "#" ;""")
+    @test ends_with_semicolon("a; #=#=# =# =#\n")
     @test ends_with_semicolon("1;")
     @test ends_with_semicolon("1;\n")
     @test ends_with_semicolon("1;\r")
     @test ends_with_semicolon("1;\r\n   \t\f")
     @test ends_with_semicolon("1;#text\n")
-    @test ends_with_semicolon("a; #=#=# =# =#\n")
+    @test ends_with_semicolon("""1;\n#text\n""")
     @test !ends_with_semicolon("begin\na;\nb;\nend")
     @test !ends_with_semicolon("begin\na; #=#=#\n=#b=#\nend")
     @test ends_with_semicolon("\na; #=#=#\n=#b=#\n# test\n#=\nfoobar\n=##bazbax\n")
+    @test ends_with_semicolon("f()= 1; # a ; 2")
+    @test ends_with_semicolon("f()= 1; # a")
+    @test !ends_with_semicolon("f()= 1; \"a\"")
+    @test !ends_with_semicolon("""("f()= 1; # a")""")
+    @test !ends_with_semicolon(""" "f()= 1; # a" """)
```

Comment:
We should also add some tests for escaped quotes, e.g.
```jl
@test !ends_with_semicolon("\"\\\";\"")
```

Reply:
Added!